testsuite/features/step_definitions/setup_steps.rb

```diff
@@ -206,16 +206,10 @@
     And I wait until I see the name of "#{host}", refreshing the page
     And I follow this "#{host}" link
     And I wait until I see "System Status" text
+    And I wait 180 seconds until the event is picked up and #{DEFAULT_TIMEOUT} seconds until the event "Apply states" is completed
+    And I wait 180 seconds until the event is picked up and #{DEFAULT_TIMEOUT} seconds until the event "Hardware List Refresh" is completed
+    And I wait 180 seconds until the event is picked up and #{DEFAULT_TIMEOUT} seconds until the event "Package List Refresh" is completed
```

Comment:
Don't we have something already to go through these 3 steps ?

"I wait until onboarding is completed" or something like that.

Reply:
```ruby
When(/^I wait until onboarding is completed for "([^"]*)"$/) do |host|
  steps %(
    When I follow the left menu "Systems > System List > All"
    And I wait until I see the name of "#{host}", refreshing the page
    And I follow this "#{host}" link
    And I wait until I see "System Status" text
  )
  if get_client_type(host) == 'traditional' && is_salt.empty?
    get_target(host).run('rhn_check -vvv')
  else
    steps %(
      And I wait 180 seconds until the event is picked up and #{DEFAULT_TIMEOUT} seconds until the event "Apply states" is completed
      And I wait 180 seconds until the event is picked up and #{DEFAULT_TIMEOUT} seconds until the event "Hardware List Refresh" is completed
      And I wait 180 seconds until the event is picked up and #{DEFAULT_TIMEOUT} seconds until the event "Package List Refresh" is completed
    )
  end
end
```
Would indeed be a good candidate.

Reply:
As you can see, I did not change these steps, I just removed the if statement for the traditional stack. So, I prefer to don't do any change on these steps on this PR.

Reply:
ok, approving then